### PR TITLE
Glasgow Class 6 - Christina Mifsud - Full Stack Assessment - Level 300

### PR DESCRIPTION
**Pull request by christina-mifsud**
Level 300 complete

### CODE REVIEW

**Comment by Dedekind561** [comment]
Hi @christina-mifsud, 
Thankyou for submitting this pull request! Your commit history looks very clear and detailed so far. Do you have a link to the deployed version of your app, please? I can't run this application locally due to missing environment variable.

**Comment by christina-mifsud** [comment]
> Hi @christina-mifsud, Thankyou for submitting this pull request! Your commit history looks very clear and detailed so far.
> 
> Do you have a link to the deployed version of your app, please? I can't run this application locally due to missing environment variables.

@Dedekind561 My apologies! I had not yet deployed it. Link found here: https://christinamifsud-fullstackassessment.onrender.com/ and in my repo 'About' section. Thanks in advance.

File: client/src/SingleVideoCard.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        setFilterVideos(updatedVideos);
+      } else {
+        console.error("Failed to delete video.");
+      }
+    } catch (error) {
+      console.error("Error deleting video:", error);
+    }
+  }
+
+  const urlId = url.split("v=")[1].substring(0, 11);
```
Comment (Dedekind561) [inline client/src/SingleVideoCard.js:44]: Hi @christina-mifsud, 
Thanks for submitting your code - I can see you are trying to debug an error at the moment. Look closely at the data you're getting back from your API ( I can see this being logged in the `console` at the moment )
 
image

Comment (Dedekind561) [reply client/src/SingleVideoCard.js:44]: If you look closely at this data does it make sense why you get a `TypeError` in the frontend when your application loads?

File: client/src/App.js [hunk 2]
```diff
@@ -1,11 +1,44 @@
...
+}
 
 function App() {
+  const [count, setCount] = useState(0); // creating counter to tell react to rebuild component every time there was a change in the db
+  const [videos, setVideos] = useState([]);
+
+  const fetchVideos = async () => {
+    try {
+      console.log("front end fetch function");
+      const response = await fetch(`${backendUrl}`);
```
Comment (Dedekind561) [inline client/src/App.js:20]: How come this isn't `${backendUrl}/videos` ?

**Review by Dedekind561 — COMMENTED** [review]
Hi @christina-mifsud, 
Thankyou for your submission - I can see lots of effort has gone into this!
First off, be sure to check you've not broken the deployed version of your application in the production environment ( although I appreciate you're submitting your work so far ! )

I've left some advice on how to identify the bug in your code though.
I hope this helps - let me know if you fix it!

File: client/src/SingleVideoCard.js (hunk 1 above)
Comment (christina-mifsud) [reply client/src/SingleVideoCard.js:44]: Thank you for your comments Mitch!
I need to add validation to check if URL is a valid youtube URL or not?

**Pull request closed without merging**